### ISSUE TITLE
auth_user:passwd?

### ISSUE THREAD

**Issue by songlong** [olivere/elastic#461]
client, err := elastic.NewClient(elastic.SetURL("http://elastic:password@47.93.79.144:9200"))
        if err != nil {
                panic(err)
        }

       connect failed?

**Comment by sundarv85** [comment]
Use SetBasicAuth option

```
client, err := elastic.NewClient(
  elastic.SetURL("http://127.0.0.1:9200", "http://127.0.0.1:9201"),
  elastic.SetBasicAuth("user", "secret"))
```

https://godoc.org/gopkg.in/olivere/elastic.v5#NewClient

**Comment by olivere** [comment]
What @sundarv85 said :-)

However, I agree that it's a good idea to be able to also set it directly via the UserInfo part of the URL.

**Comment by songlong** [comment]
thanks
I think go and python the same.
but not.

**Comment by olivere** [comment]
Fixed in 5.0.25.
